src/cms/templates/events/event_list.html

```diff
@@ -27,23 +41,30 @@ <h2 class="heading font-normal mb-2">{% trans 'Events' %}</h2>
         <div class="flex flex-wrap">
             <form class="table-search relative w-1/2">
                 <i data-feather="search" class="absolute"></i>
-                <input type="search" placeholder="{% trans Search %}" class="h-full pl-10 pr-4 rounded shadow">
+                <input type="search" placeholder="{% trans 'Search' %}" class="h-full pl-10 pr-4 rounded shadow">
             </form>
-            {% if region.default_language == language %}
             <div class="w-1/2 flex flex-wrap justify-end">
-                <a href="{% url 'new_event' region_slug=region.slug language_code=language.code %}"
-                   class="bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline">
-                    {% trans 'Create event' %}
-                </a>
+                <button id="filter-toggle"
+                        class="bg-red-500 hover:bg-red-600 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline mr-3">
+                    {% trans 'Filter events' %}
+                </button>
+                {% if region.default_language == language %}
+                    <a href="{% url 'new_event' region_slug=region.slug language_code=language.code %}"
+                       class="bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline">
+                        {% trans 'Create event' %}
+                    </a>
+                {% endif %}
             </div>
-            {% endif %}
         </div>
     </div>
+    <div id="filter-form-container" class="hidden w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white">
```

Comment:
```suggestion
    <div id="filter-form-container" class="{% if not filter_form.data %}hidden{% endif %} w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white">
```
I think it could be useful to leave the filter div expanded after the filters have been applied, but if you prefer it the way it is, just keep it.

Reply:
`filter_form.data` does not work because a `POST` request with an empty form (i.e. applying the filter after resetting it) also sends data even though no filtering was actually done. I used `filter_form.has_changed` here, this works like a charm.
Thanks for the suggestion!